lusidtools/cocoon/async_tools.py

```diff
@@ -47,9 +65,15 @@ def run_in_executor(f):
     """
     Passes a synchronous & blocking function off to another thread so that it can be awaited
 
-    :param f: The function to transform into an awaitable
+    Parameters
+    ----------
+    f
+        The function to transform into an awaitable
 
-    :return: An awaitable version of the function with the execution delegated to another thread
+    Returns
+    -------
+    callable
+        An awaitable version of the function with the execution delegated to another thread
```

Comment:
Should this be "inner : callable"? "callable" is the type.

Reply:
Ah yes, I'll make this change now 👍 

---------------------------------------------
A lot of this is just converting directly from the old docstring, which in this case was missing the "inner" also which is why it was left like this

when no name is specified then the type can be used alone
```
    """Example function with PEP 484 type annotations.

    The return type must be duplicated in the docstring to comply
    with the NumPy docstring style.

    Parameters
    ----------
    param1
        The first parameter.
    param2
        The second parameter.

    Returns
    -------
    bool
        True if successful, False otherwise.

    """
```
```
"""Sum numbers in an array.

Parameters
----------
values : iterable
   Python iterable whose values are summed.

Returns
-------
sum : `float`
   Sum of ``values``.
"""
```